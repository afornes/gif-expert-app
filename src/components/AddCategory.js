import PropTypes from 'prop-types';
import React, { useState } from 'react'

const AddCategory = ({setCategories}) => {

    const [inputValue, setIntpuValue] = useState('');

    const onInputChange = (e) => {
        setIntpuValue(e.target.value);
    };

    const onEnterPressed = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setIntpuValue('');
        }
    }

    return (
        <form onSubmit={onEnterPressed}>
            <input 
                type='text'
                value={ inputValue }
                onChange={ onInputChange }
                placeholder='Buscar...'
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
