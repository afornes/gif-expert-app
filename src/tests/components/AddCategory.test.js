import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Tests for AddCategory Component', () => { 
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    });
    
    test('should return valid snapshot', () => { 
        
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should change textfield', () => { 
        
        const input = wrapper.find('input');
        const value = 'Hello world';
        
        input.simulate('change', { target: { value } });        
    });
    
    test('should not post on submit', () => { 
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('should call setCategories and clean the textfield', () => { 
        
        const input = wrapper.find('input');
        const value = 'Monster Hunter';
        
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('');
    });
});