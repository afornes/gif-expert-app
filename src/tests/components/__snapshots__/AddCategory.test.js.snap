// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AddCategory Component should return valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar..."
    type="text"
    value=""
  />
</form>
`;
